File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/auth/AuthenticationProcessTemplateParameters.kt
package com.icure.cardinal.sdk.auth

import kotlinx.serialization.Serializable

/**
 * Optional parameters for the authentication process template
 */
@Serializable
data class AuthenticationProcessTemplateParameters(
	/**
	 * First name of the user executing the authentication process.
	 */
	val firstName: String? = null,
	/**
	 * Last name of the user executing the authentication process.
	 */
	val lastName: String? = null
)

File: dart-wrapper/cardinal_sdk/lib/auth/authentication_process_template_parameters.dart
class AuthenticationProcessTemplateParameters {
  final String? firstName;
  final String? lastName;

  const AuthenticationProcessTemplateParameters({
    this.firstName,
    this.lastName
  });

  static Map<String, dynamic> encode(AuthenticationProcessTemplateParameters value) {
    return {
      "firstName": value.firstName,
      "lastName": value.lastName
    };
  }
}